### ISSUE TITLE
Installation failure with GPU:  DNN library is not found

### ISSUE THREAD

**Issue by mattiasmar** [google/flax#1769]
Provide as much information as possible. At least, this should include a description of your issue and steps to reproduce the problem. If possible also provide a summary of what steps or workarounds you have already tried.

### Problem you have encountered:
The installation instructions are not complete.
Installing Flax in a TF docker container according instructions are not enough to avoid  `RuntimeError: UNKNOWN: Failed to determine best cudnn convolution algorithm for %cudnn-conv ...` in `flax/linen/linear.py` when executing Flax's example programs.

### What you expected to happen:
Flax example program to complete without runtime errors.

### Logs, error messages, etc:
```
Original error: UNIMPLEMENTED: DNN library is not found.

To ignore this failure and try to use a fallback algorithm (which may have suboptimal performance), use XLA_FLAGS=--xla_gpu_strict_conv_algorithm_picker=false.  Please also file a bug for the root cause of failing autotuning.

The stack trace below excludes JAX-internal frames.
The preceding is the original exception that occurred, unmodified.

--------------------

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "train_test.py", line 46, in test_cnn
    output, variables = train.CNN().init_with_output(rng, inputs)
  File "/tf/flax/examples/mnist/train.py", line 39, in __call__
    x = nn.Conv(features=32, kernel_size=(3, 3))(x)
  File "/usr/local/lib/python3.8/dist-packages/flax/linen/linear.py", line 282, in __call__
    y = lax.conv_general_dilated(
RuntimeError: UNKNOWN: Failed to determine best cudnn convolution algorithm for:
%cudnn-conv = (f32[1,28,28,32]{2,1,3,0}, u8[0]{0}) custom-call(f32[1,28,28,3]{2,1,3,0} %copy.3, f32[3,3,3,32]{1,0,2,3} %copy.4), window={size=3x3 pad=1_1x1_1}, dim_labels=b01f_01io->b01f, custom_call_target="__cudnn$convForward", metadata={op_type="conv_general_dilated" op_name="jit(conv_general_dilated)/conv_general_dilated[\n  batch_group_count=1\n
dimension_numbers=ConvDimensionNumbers(lhs_spec=(0, 3, 1, 2), rhs_spec=(3, 2, 0, 1), out_spec=(0, 3, 1, 2))\n  feature_group_count=1\n  lhs_dilation=(1, 1)\n  lhs_shape=(1, 28, 28, 3)\n  padding=((1, 1), (1, 1))\n  precision=None\n  preferred_element_type=None\n  rhs_dilation=(1, 1)\n  rhs_shape=(3, 3, 3, 32)\n  window_strides=(1, 1)\n]" source_file="/usr/local/lib/python3.8/dist-packages/flax/linen/linear.py" source_line=282}, backend_config="{\"conv_result_scale\":1,\"activation_mode\":\"0\",\"side_input_scale\":0}"

Original error: UNIMPLEMENTED: DNN library is not found.

To ignore this failure and try to use a fallback algorithm (which may have suboptimal performance), use XLA_FLAGS=--xla_gpu_strict_conv_algorithm_picker=false.  Please also file a bug for the root cause of failing autotuning.
```

### Steps to reproduce:
1. Start TF docker container on a GPU host:  `docker run  --gpus all -it --rm  tensorflow/tensorflow:latest-gpu-jupyter  /bin/bash`
2. Inside the docker container:
```
export PATH=/usr/local/cuda-11.2/bin/:$PATH
export LD_LIBRARY_PATH=/usr/local/cuda-11.2/lib64:$LD_LIBRARY_PATH
pip install --upgrade pip
pip install jax[cuda11_cudnn82] -f https://storage.googleapis.com/jax-releases/jax_releases.html
pip install clu ml-collections tensorflow_datasets
git clone https://github.com/google/flax.git
cd flax/examples/mnist
python train_test.py
```

**Comment by mattiasmar** [comment]
When using the alternative jax installation:
`pip install jax[cuda11_cudnn805] -f https://storage.googleapis.com/jax-releases/jax_releases.html`

A different error appears:
```
2022-01-09 14:46:52.015253: E external/org_tensorflow/tensorflow/stream_executor/cuda/cuda_dnn.cc:4584] Disabling cuDNN frontend for the following convolution:
  input: {count: 1 feature_map_count: 3 spatial: 28 28  value_min: 0.000000 value_max: 0.000000 layout: BatchDepthYX}
  filter: {output_feature_map_count: 32 input_feature_map_count: 3 layout: OutputInputYX shape: 3 3 }
  {zero_padding: 1 1  pad_alignment: default filter_strides: 1 1  dilation_rates: 1 1 }
  ... because the current cuDNN version does not support it.
2022-01-09 14:46:54.106367: E external/org_tensorflow/tensorflow/stream_executor/cuda/cuda_dnn.cc:4584] Disabling cuDNN frontend for the following convolution:
  input: {count: 1 feature_map_count: 32 spatial: 14 14  value_min: 0.000000 value_max: 0.000000 layout: BatchDepthYX}
  filter: {output_feature_map_count: 64 input_feature_map_count: 32 layout: OutputInputYX shape: 3 3 }
  {zero_padding: 1 1  pad_alignment: default filter_strides: 1 1  dilation_rates: 1 1 }
  ... because the current cuDNN version does not support it.
[       OK ] TrainTest.test_cnn
[ RUN      ] TrainTest.test_train_and_evaluate
[  FAILED  ] TrainTest.test_train_and_evaluate
======================================================================
ERROR: test_train_and_evaluate (__main__.TrainTest)
TrainTest.test_train_and_evaluate
Tests training and evaluation code by running a single step.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "train_test.py", line 59, in test_train_and_evaluate
    flax_root_dir = pathlib.Path(__file__).parents[2]
  File "/usr/lib/python3.8/pathlib.py", line 621, in __getitem__
    raise IndexError(idx)
IndexError: 2
```

**Comment by mattiasmar** [comment]
These errors were due to incompatibility between the preinstalled cudnn of the docker image and the jax wheel.
Resolved by using the docker image `nvidia/cuda:11.4.1-cudnn8-devel-ubuntu18.04` and installing jax with:
`pip install --upgrade "jax[cuda]" -f https://storage.googleapis.com/jax-releases/jax_releases.html`
